ludwig/data/preprocessing.py

```diff
@@ -295,247 +298,140 @@ def load_metadata(metadata_file_path):
             preprocessing_params=preprocessing_params,
             random_seed=random_seed
         )
-    elif data_hdf5 is not None or data_train_hdf5 is not None:
-        return preprocess_for_training_by_type(
-            model_definition,
-            'hdf5',
-            all_data_fp=data_hdf5,
-            train_fp=data_train_hdf5,
-            validation_fp=data_validation_hdf5,
-            test_fp=data_test_hdf5,
-            train_set_metadata_json=train_set_metadata_json,
+
+    elif data_format in CSV_FORMATS:
+        (
+            training_set,
+            test_set,
+            validation_set,
+            training_set_metadata
+        ) = _preprocess_csv_for_training(
+            features,
+            dataset,
+            training_set,
+            validation_set,
+            test_set,
+            training_set_metadata=training_set_metadata,
             skip_save_processed_input=skip_save_processed_input,
             preprocessing_params=preprocessing_params,
             random_seed=random_seed
         )
-    else:
-        raise ValueError('Invalid type of data provided or Invalid usage of '
-                         'datasets. Please review your command')
 
+    elif data_format in HDF5_FORMATS:
+        if not dataset:
+            raise ValueError('When providing HDF5 data, '
+                             'data must not be None and '
+                             'should be the path to a HDF5 file containing '
+                             'train, validation and test splits.')
 
-def preprocess_for_training_by_type(
-        model_definition,
-        data_type,
-        all_data_fp=None,
-        train_fp=None,
-        validation_fp=None,
-        test_fp=None,
-        all_data_df=None,
-        train_df=None,
-        validation_df=None,
-        test_df=None,
-        train_set_metadata_json=None,
-        skip_save_processed_input=False,
-        preprocessing_params=default_preprocessing_parameters,
-        random_seed=default_random_seed
-):
-    if all_data_fp is not None and train_fp is not None:
-        raise ValueError('Use either one file for all data or 3 files for '
-                         'train, test and validation')
-
-    if data_type not in ['hdf5', 'csv', 'pandas']:
-        raise ValueError('Invalid type of data provided')
+        if not training_set_metadata:
+            raise ValueError('When providing HDF5 data, '
+                             'training_set_metadata must not be None.')
 
-    features = (model_definition['input_features'] +
-                model_definition['output_features'])
+        logger.info('Using full hdf5 and json')
+        training_set, test_set, validation_set = load_hdf5(
+            dataset,
+            model_definition['input_features'],
+            model_definition['output_features'],
+            shuffle_training=True
+        )
 
-    data_hdf5_fp = None
+    elif data_format in DICT_FORMATS:
+        if dataset:
```

Comment:
Nice way to condense this part:

```
dataset = dataset or pd.DataFormat(dataset)
```

Reply:
Not sure, my logic here is: if dataset/training_set/etc. are None, keep them none, if the are not None, then convert them to dataframes. I guess that if I do `dataset = dataset or pd.DataFormat(dataset)`, if it's None, I'll get an empy dataframe, and when it's not None, I'll get the dictionary/